internal/notifier/google_pubsub_topic_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2023 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	eventv1 "github.com/fluxcd/pkg/apis/event/v1beta1"
+)
+
+type googlePubSubTestCase struct {
+	topicName         string
+	expectedProjectID string
+	expectedTopicID   string
+	expectedParseErr  error
+	event             eventv1.Event
+	expectedMessage   string
+	publishErr        error
+	expectedPostErr   error
+	expectedCloseErr  error
+}
+
+func (tt *googlePubSubTestCase) Publish(ctx context.Context, topicID string, msg []byte) (serverID string, err error) {
+	if topicID != tt.expectedTopicID {
+		return "", fmt.Errorf("topicID does not match expected. want %v, got %v", tt.expectedTopicID, topicID)
+	}
```

Comment:
We already require Gomega in this module so we could also do
```suggestion
	g.Expect(topicID).To(Equal(tt.expectedTopicID), "unexpected topicID")
```
I'll leave this to you but I believe this to be much more readable, apart from being shorter, too.

Reply:
The thing is that this struct doesn't have the `g` instance. But now it does :grin: Thanks!